### PR TITLE
feat: triplet margin loss added to functional experimental

### LINKED ISSUE #27204: triplet_margin_loss

### PR DESCRIPTION
**Pull request by BhumikaChauhan07**
# PR Description
Changes Made
Added a new function triplet_margin_loss to the ivy library, which computes the triplet margin loss. This loss is commonly used in triplet loss scenarios, such as in siamese networks for learning embeddings.

Purpose
The purpose of this addition is to provide users with a convenient and efficient way to compute triplet margin loss, allowing for better support of applications involving metric learning and similarity-based tasks.

## Related Issue
Closes #27204

## Checklist

- [X] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [ ] Did you follow the steps we provided?

### CODE REVIEW

**Comment by NiteshK84** [comment]
Hi @BhumikaChauhan07,
Please resolve the merge conflicts.

**Comment by BhumikaChauhan07** [comment]
Hello @NiteshK84, I'm on it. Thanks.

**Comment by NiteshK84** [comment]
Hi @BhumikaChauhan07,
you have not implemented the unit test_function
please implement it in this path "/workspaces/ivy/ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py"

**Comment by BhumikaChauhan07** [comment]
Ok I'll work upon it.

**Comment by BhumikaChauhan07** [comment]
Hi @NiteshK84,
I've implemented the respective test function. Regards.

**Review by NiteshK84 — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py [hunk 1]
```diff
@@ -449,3 +449,39 @@ def test_soft_margin_loss(
         target=target[0],
         reduction=reduction,
     )
+
+
+# triplet_loss
+@handle_test(
+    fn_tree="functional.ivy.experimental.triplet_loss",
+    dtype_and_inputs=_triplet_loss_input(),
```
[the remark this answers is not in the document]
Comment (BhumikaChauhan07) [reply ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py:457]: Okay , on it.

**Comment by Ishticode** [comment]
Thank you very much for the PR. Closing this due to lack of activity with changes requested remaining unresolved for over 2 months. Please feel free to reopen if you would like to continue working on this. :)

**Pull request closed without merging**